plugins/dicom_viewer/server/__init__.py

```diff
@@ -46,6 +46,65 @@ def getDicom(self, item, filters, force):
                 f['dicom'] = dicom
         return files
 
+    @access.public(scope=TokenScope.DATA_READ)
+    @autoDescribeRoute(
+        Description('Get and store common DICOM metadata, if any, for all files in the item.')
+        .modelParam('id', 'The item ID',
+                    model='item', level=AccessType.READ, paramType='path')
+        .errorResponse('ID was invalid.')
+        .errorResponse('Read permission denied on the item.', 403)
+    )
+    def makeDicomItem(self, item):
+        """
+        Try to convert an existing item into a "DICOM item", which contains a
+        "dicomMeta" field with DICOM metadata that is common to all DICOM files.
+        """
+        files = filter(_isDicomFile, ModelImporter.model('item').childFiles(item))
+        if not files:
+            return
+        dicomFileMeta = parse_file(file[0])
+        item['dicomMeta'] = dicomFileMeta
+        for additionalFile in files[1:]:
+            additionalFileMetadata = parse_file(additionalFile)
+            item['dicomMeta'] = removeUniqueMetadata(dicomFileMeta, additionalFileMetadata)
+        return ModelImporter.model('item').save(item)
+
+
+def removeUniqueMetadata(dicomMeta, additionalMeta):
+    dicomValues = []
```

Comment:
Python `dict`s have random access, so I think your implementation here can be significantly simplified. Let's discuss this directly.